db/src/metadata.rs

```diff
@@ -105,7 +105,38 @@ impl MetadataReport {
 /// Returns a report summarizing the mutations that were applied.
 pub fn update_attribute_map_from_entid_triples<A, R>(attribute_map: &mut AttributeMap, assertions: A, retractions: R) -> Result<MetadataReport>
     where A: IntoIterator<Item=(Entid, Entid, TypedValue)>,
-          R: IntoIterator<Item=(Entid, Entid, TypedValue)> {
+          R: Iterator<Item=(Entid, Entid, TypedValue)> {
+
+    // Process retractions of schema attributes first. It's allowed to retract a schema attribute
```

Comment:
For excision, I built a little EAV trie -- see https://github.com/mozilla/mentat/commit/03dd4a37a35602c5ee0256be8504f9672393d744#diff-929f073706d06052e936ffa3c71e523aR217.  I'd like to standardize these types of filtering and searching operations around `AEVTrie` and `EAVTrie`.  Take a look.  I think it would make this a little more pleasant.

Reply:
I tied going down that path initially, but couldn't make the ownership and lifetimes work.